cmd/snap/cmd_model.go

```diff
@@ -0,0 +1,92 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"text/tabwriter"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/i18n"
+)
+
+var (
+	shortModelHelp = i18n.G("Get the active model for this device")
+	longModelHelp  = i18n.G(`TODO`)
+)
+
+type cmdModel struct {
+	waitMixin
+	Serial    bool `long:"serial"`
+	Verbose   bool `long:"verbose"`
+	Assertion bool `long:"assertion"`
+}
+
+func init() {
+	cmd := addCommand("model",
+		shortModelHelp,
+		longModelHelp,
+		func() flags.Commander {
+			return &cmdModel{}
+		}, nil, []argDesc{},
+	)
+	cmd.hidden = true
+}
+
+func (x *cmdModel) Execute(args []string) error {
+	var assertion asserts.Assertion
+	var err error
+	if x.Serial {
+		assertion, err = x.client.CurrentSerialAssertion()
+	} else {
+		assertion, err = x.client.CurrentModelAssertion()
+	}
+
+	if err != nil {
+		return err
+	}
+
+	// --assertion means output the raw assertion
+	if x.Assertion {
+		return asserts.NewEncoder(Stdout).Encode(assertion)
```

Comment:
I suspect doing asserts.Encode separely and then print might be slightly cleaner in this case

Reply:
okay, done